rust-create-cascade/src/clubcard_helper.rs

```diff
@@ -62,21 +62,22 @@ impl FilterBuilder for ClubcardBuilder<4, CRLiteBuilderItem> {
         let ribbons: Vec<ApproximateRibbon<4, CRLiteBuilderItem>> =
             list_issuer_file_pairs(revoked_dir, known_dir)
                 .par_iter()
-                .map(|pair| {
-                    if let (issuer, Some(revoked_file), known_file) = pair {
-                        let issuer_bytes =
-                            decode_issuer(issuer.to_str().expect("non-unicode issuer string"));
-                        Some(clubcard_do_one_issuer(
-                            self,
-                            &issuer_bytes,
-                            RevokedSerialAndReasonIterator::new(revoked_file, reason_set),
-                            KnownSerialIterator::new(known_file),
-                        ))
-                    } else {
-                        None
-                    }
+                .map(|(issuer, maybe_revoked_file, known_file)| {
+                    let issuer_bytes =
+                        decode_issuer(issuer.to_str().expect("non-unicode issuer string"));
+                    let revoked_serials_and_reasons = match maybe_revoked_file {
+                        Some(revoked_file) => {
+                            RevokedSerialAndReasonIterator::new(revoked_file, reason_set)
+                        }
+                        None => Default::default(), // empty iterator
```

Comment:
I'm a little concerned that the default `RevokedSerialAndReasonIterator` may have a different `reason_set` here, in terms of potential future changes. What do you think about passing `maybe_revoked_file` through to `RevokedSerialAndReasonIterator::new` instead of using `Default`?

Reply:
That's reasonable. Changing `new` would require a bunch of changes in other parts of the code. So I added an explicit constructor for empty `RevokedSerialAndReasonIterator`s that takes a reason set.

Reply:
Ok!